lib/ptracer.h

```diff
@@ -0,0 +1,485 @@
+
+        /* Set stack container to registry .*/
+        lua_setfield(L, LUA_REGISTRYINDEX, PALLENE_TRACER_CONTAINER_ENTRY);
+
+        /* The debug traceback fn. */
+        lua_register(L, "pallene_tracer_debug_traceback", pallene_tracer_debug_traceback);
+
+        /* Push the finalizer object in the stack. */
+        lua_getfield(L, LUA_REGISTRYINDEX, PALLENE_TRACER_FINALIZER_ENTRY);
+    } else {
+        cont = lua_touserdata(L, -1);
+        lua_getfield(L, LUA_REGISTRYINDEX, PALLENE_TRACER_FINALIZER_ENTRY);
+    }
+
+    return cont;
+}
+
+/* Pushes a frame to the stack. The frame structure is self-managed for every function. */
+void pallene_tracer_frameenter(lua_State *L, pt_cont_t *cont, pt_frame_t *restrict frame) {
+    /* Have we ran out of stack entries? */
+    if(luai_unlikely(cont->count + 1 >= PALLENE_TRACER_MAX_CALLSTACK)) {
+        pallene_tracer_runtime_callstack_overflow_error(L);
+    }
+
+    cont->stack[cont->count++] = *frame;
+}
+
+/* Sets line number to the topmost frame in the stack. */
+void pallene_tracer_setline(pt_cont_t *cont, int line) {
+    if(luai_likely(cont->count != 0))
+        cont->stack[cont->count - 1].line = line;
+}
+
+/* Removes the last frame from the stack. */
+void pallene_tracer_frameexit(pt_cont_t *cont) {
+    cont->count -= (cont->count > 0);
+}
+
+/* Helper macro specific to this function only :). */
+#define DBG_PRINT() _pallene_tracer_dbg_print(buf, &ellipsis, &pframes, nframes)
+/* Pallene Tracer explicit traceback function to show Pallene call-stack
+   tracebacks. */
+int pallene_tracer_debug_traceback(lua_State *L) {
+    lua_getfield(L, LUA_REGISTRYINDEX, PALLENE_TRACER_CONTAINER_ENTRY);
+    pt_cont_t *cont = (pt_cont_t *) lua_touserdata(L, -1);
+    pt_frame_t *stack = cont->stack;
+    /* The point where we are in the Pallene stack. */
+    int index = cont->count - 1;
+    lua_pop(L, 1);
+
+    /* Max number of white and black frames. */
+    int mwhite, mblack;
+    _pallene_tracer_countframes(cont, &mwhite, &mblack);
+    /* Max levels of Lua stack. */
+    int mlevel = _pallene_tracer_countlevels(L);
+
+    /* Total frames we are going to print. */
+    /* Black frames are used for switching and we will start from
+       Lua stack level 1. */
+    int nframes = mlevel + mwhite - mblack - 1;
+    /* Amount of frames printed. */
+    int pframes = 0;
+    /* Should we print ellipsis? */
+    bool ellipsis = nframes > (PALLENE_TRACEBACK_TOP_THRESHOLD
+        + PALLENE_TRACEBACK_BOTTOM_THRESHOLD);
+
+    /* Buffer to store for a single frame line to be printed. */
+    char buf[1024];
+
+    const char *message = lua_tostring(L, 1);
+    fprintf(stderr, "Runtime error: %s\nStack traceback:\n", message);
+
+    lua_Debug ar;
+    int top = lua_gettop(L);
+    int level = 1;
+
+    while(lua_getstack(L, level++, &ar)) {
+        /* Get additional information regarding the frame. */
+        lua_getinfo(L, "Slntf", &ar);
```

Comment:
Do we need `t` here?
IIRC, `t` only tells us where the call was a tail call or not.

Maybe we can also modify the comment like this:
```c
/* Get athe following dditional information regarding the frame: .... */
```